src/modules/dashboard/components/ExpenditurePage/Split/SplitEqual/SplitEqual.tsx

```diff
@@ -91,6 +92,31 @@ const SplitEqual = ({ colony, sidebarRef }: Props) => {
     return isNaN(result) ? 0 : result;
   }, [amount, recipientsCount]);
 
+  useEffect(() => {
+    setValue(
+      recipients.map((recipient) => ({
+        ...recipient,
+        amount: { value: calculatedAmount, tokenAddress: amount.tokenAddress },
+      })),
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [calculatedAmount]);
+
+  const onTokenChange = useCallback(
+    (value: string) => {
+      setValue(
+        recipients.map((rec) => ({
+          ...rec,
+          amount: {
+            value: rec.amount?.value,
+            tokenAddress: value,
+          },
+        })),
+      );
+    },
+    [recipients, setValue],
```

Comment:
Given that setValue is recreated on every render, there's no point in using `useCallback` here -- or, remove from dependencies array.